tests/test_env.py

```diff
@@ -1,20 +1,74 @@
 from __future__ import annotations
 
+import re
 from pathlib import Path
 
 import pytest
+from _pytest.fixtures import FixtureRequest
 
 
-@pytest.fixture()
-def example(testdir: pytest.Testdir) -> pytest.Testdir:
-    src = Path(__file__).parent / "example.py"
-    dest = Path(str(testdir.tmpdir / "test_example.py"))
-    dest.symlink_to(src)
-    return testdir
+@pytest.mark.parametrize(
+    ("env", "ini", "expected_env"),
+    [
+        pytest.param(
+            {},  # empty dict means no preexisting environment variables
+            "[pytest]\nenv = MAGIC=alpha",
+            {"MAGIC": "alpha"},
+            id="new key, add to env",
+        ),
+        pytest.param(
+            {"MAGIC": "alpha"},
+            "[pytest]\nenv = MAGIC=beta",
+            {"MAGIC": "beta"},
+            id="key exists in env, overwrite",
+        ),
+        pytest.param(
+            {"MAGIC": "alpha"},
+            "[pytest]\nenv = D:MAGIC=beta",
+            {"MAGIC": "alpha"},
+            id="D exists, original val kept",
+        ),
+        pytest.param(
+            {"PLANET": "world"},
+            "[pytest]\nenv = MAGIC=hello_{PLANET}",
+            {"MAGIC": "hello_world"},
+            id="curly exist, interpolate var",
+        ),
+        pytest.param(
+            {"PLANET": "world"},
+            "[pytest]\nenv = R:MAGIC=hello_{PLANET}",
+            {"MAGIC": "hello_{PLANET}"},
+            id="R exists, not interpolate var",
+        ),
+        pytest.param(
+            {"MAGIC": "a"},
+            "[pytest]\nenv = R:MAGIC={MAGIC}b\n D:MAGIC={MAGIC}c\n MAGIC={MAGIC}d",
+            {"MAGIC": "{MAGIC}bd"},
+            id="incremental interpolation",
+        ),
+        pytest.param(
+            {"MAGIC": "alpha"},
+            "[pytest]\nenv = D:R:MAGIC=beta",
+            {"MAGIC": "alpha"},
+            id="two flags",
+        ),
+        pytest.param(
+            {"MAGIC": "alpha"},
+            "[pytest]\nenv = R:D:MAGIC=beta",
+            {"MAGIC": "alpha"},
+            id="two flags, reversed",
+        ),
+    ],
+)
+def test_env(testdir: pytest.Testdir, env: dict[str, str], ini: str, expected_env: dict[str, str],
+             request: FixtureRequest) -> None:
+    for env_var, val in env.items():
+        testdir.monkeypatch.setenv(env_var, val)
+    testdir.monkeypatch.setenv("_TEST_ENV", repr(expected_env))
 
+    test_name = re.sub(r'\W|^(?=\d)', '_', request.node.callspec.id).lower()
```

Comment:
Adding `test_name` to pytest logs (through filename) improves readability, because it is available in more places rather than just fixture
```
FAILED               [ 50%]
============================= test session starts ==============================
platform darwin -- Python 3.7.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /private/var/folders/0r/4z0gbyx13wvblzzp5sk7p9980000gn/T/pytest-of-elijas/pytest-130/test_env3, configfile: pytest.ini
plugins: env-0.1.dev23+g23b269e.d20221023
collected 1 item

test_curly_exist__interpolate_var.py F                                   [100%]

=================================== FAILURES ===================================
___________________________________ test_env ___________________________________

    def test_env() -> None:
        for key, value in ast.literal_eval(os.environ["_TEST_ENV"]).items():
>           assert os.environ[key] == value, key
E           AssertionError: MAGIC
E           assert 'hello_worqld' == 'hello_world'
E             - hello_world
E             + hello_worqld
E             ?          +

test_curly_exist__interpolate_var.py:9: AssertionError
=========================== short test summary info ============================
FAILED test_curly_exist__interpolate_var.py::test_env - AssertionError: MAGIC
============================== 1 failed in 0.04s ===============================
```